### ISSUE TITLE
Moloch URL parts hardcoded

### ISSUE THREAD

**Issue by mkruk-safecomp** [cuckoosandbox/cuckoo#2309]
##### My issue is:
The URLs generated in web interface consist of non-configurable hardcoded parts.

##### My Cuckoo version and operating system are:

I'm using Cuckoo under CentOS7 and my cuckoo version is 2.0.5

##### This can be reproduced by:

Enabling moloch integration

##### The log, error, files etc can be found at:

The relevant file here is cuckoo/web/analysis/view.py
It contains:
```
def moloch(request, **kwargs):
    if not config("reporting:moloch:enabled"):
        return view_error(request, "Moloch is not enabled!")

    query = []
    for key, value in kwargs.items():
        if value and value != "None":
            query.append(moloch_mapper[key] % value)

    if ":" in request.get_host():
        hostname = request.get_host().split(":")[0]
    else:
        hostname = request.get_host()

    if config("reporting:moloch:insecure"):
        url = "http://"
    else:
        url = "https://"

    url += "%s:8005/?%s" % (
        config("reporting:moloch:host") or hostname,
        urllib.urlencode({
            "date": "-1",
            "expression": " && ".join(query),
        }),
    )
    return redirect(url)

```

It assumes that the destination moloch installation is running on default port (which is not always the case) and uses default base URL, which is not true as well in some setups.
Real-life scenario - my cuckoo/moloch infrastructure is behind a proxy so even though moloch does work on http://localhost:8005/ I connect from remote using https://proxy.hostname/moloch/ (the proxy also works as SSL terminator).

So the proper approach IMHO would be to make whole base URL configurable. For now, even if I set moloch host in cuckoo configuration to something other than cuckoo hostname, I'm at most getting
https://proxy.hostname:8005/?query_string instead of required https://proxy.hostname/moloch/?query_string

**Comment by RicoVZ** [comment]
Hi mkrsfcmp,

Thanks for creating an issue.

You are right, this does seem a bit silly. This should be fixed if the port would come from the config?
It would probably make more sense if the default port would only be used if no port was specified in the config.

**Comment by RicoVZ** [comment]
@hackdefendr What have you entered in the `host` field in `reporting.conf`?

It should only contain the IP/hostname. So `host = xx.xx.xxx.xxx`, not including a port or a protocol.

**Comment by hackdefendr** [comment]
I don't have anything in the host field, because my Moloch and Cuckoo are on the same IP.  The comment in the reporting.conf states to only put the IP in if Moloch is on a different IP.  Unless I read it wrong, but 99% I'm sure it says that.

**Comment by RicoVZ** [comment]
Yes, you are right, That is the correct way. :smile: I wanted to make sure a faulty host entry was not the cause.

Mhm, odd stuff. I will have to try and reproduce this. Can you create a new issue for this please, and tag me in it? :smile:  I would like to keep each possible bug in a seperate one.
